bin/steps/sqlite3

```diff
@@ -66,14 +66,21 @@ buildpack_sqlite3_install() {
 
   # python version check
   if python_sqlite3_check "$PYTHON_VERSION"; then
-    puts-step "Installing SQLite3"
-
-    if sqlite3_install "$BUILD_DIR/.heroku/python" ; then
-      echo "Sqlite3 successfully installed."
-      mcount "success.python.sqlite3"
-    else
-      echo "Sqlite3 failed to install."
-      mcount "failure.python.sqlite3"
+    # only install if the sqlite3 version has changed
+    if [ "$INSTALLED_SQLITE3_VERSION" != "$SQLITE3_VERSION" ]; then
+      puts-step "Installing SQLite3"
+
+      if sqlite3_install "$BUILD_DIR/.heroku/python" "$SQLITE3_VERSION" ; then
+        echo "Sqlite3 successfully installed."
+        mcount "success.python.sqlite3"
+      else
+        echo "Sqlite3 failed to install."
+        mcount "failure.python.sqlite3"
+      fi
+
+      # save version installed
+      mkdir -p "$CACHE_DIR/.heroku/"
+      echo "$SQLITE3_VERSION" > "$CACHE_DIR/.heroku/python-sqlite3-version"
```

Comment:
I would think we would also want to use what is on **master** here as well. This pin was fixed in: https://github.com/heroku/heroku-buildpack-python/commit/ffc7683c05c924d345b43c6ad1c5ea8b4a5a5fb1